### PR TITLE
effect flags

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

File: src/effects/big_snowflakes.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ void big_snowflakes_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3) {
     bp.init = big_snowflakes_init;
     bp.update = big_snowflakes_update;
     bp.renderWorld = big_snowflakes_render;
-    bp.unk_14 = 0;
+    bp.renderUI = 0;
```
Comment (ethteck) [inline src/effects/big_snowflakes.c:27]: these renderUI = 0 should all be renderUI = NULL. it looks like we were inconsistent

**Review by ethteck — CHANGES_REQUESTED** [review]

**Review by nanaian — CHANGES_REQUESTED** [review]
And the NULL thing. Cool PR

File: src/effects/lens_flare.c [hunk 2]
```diff
@@ -106,53 +108,55 @@ void lens_flare_render(EffectInstance* effect) {
...
-    idx = data->unk_30 * 3;
+    alpha = data->smallFlareAlpha;
+    idx = data->lifetime * 3;
 
-    if (unk_00 == 0) {
-        shim_guScaleF(sp58, data->unk_20, data->unk_20, data->unk_20);
-        shim_guMtxCatF(sp58, sp98, sp18);
-        shim_guMtxF2L(sp18, &gDisplayContext->matrixStack[gMatrixListPos]);
+    // small twinkling shine which changes color
+    if (type == 0) {
```
[the remark this answers is not in the document]
Comment (HailSanta) [reply src/effects/lens_flare.c:130]: Every effect has a type provided in the first arg. Some of the definitely deserve an enum, others I'm more ambivalent about. Many of them only have a single type, others have just two and could be represented as a bool. For those with >2 types, choosing (reasonably succinct) names for them all is the hard part. A couple enums are already defined in effects.h, just before the relevant FXData.

File: src/effects.c [hunk 3]
```diff
@@ -340,47 +347,48 @@ EffectInstance* create_effect_instance(EffectBlueprint* effectBp) {
...
+        effectGraphics->update = effectBp->update;
+        if (effectGraphics->update == NULL) {
+            effectGraphics->renderWorld = stub_effect_delegate;
         }
 
-        curEffect->renderWorld = effectBp->renderWorld;
-        if (curEffect->renderUI == NULL) {
-            curEffect->renderUI = stub_effect_delegate;
+        effectGraphics->renderWorld = effectBp->renderWorld;
+        // @bug? null check for renderUI instead of renderWorld
```
[the remark this answers is not in the document]
Comment (HailSanta) [reply src/effects.c:372]: The code is written as expected until just after this line, where it does a NULL check for renderUI, followed by binding renderUI and performing the NULL check for renderUI again. The one bearing the comment should be checking renderWorld to match the binding on the previous line.

**Pull request merged**